medallion/backends/mongodb_backend.py

```diff
@@ -78,27 +91,29 @@ def server_discovery(self):
         info = list(collection.aggregate(pipeline))[0]
         return info
 
+    @catch_mongodb_error
     def get_collections(self, api_root):
-        # TODO: Handle if mongodb is not available
         api_root_db = self.client[api_root]
         collection_info = api_root_db["collections"]
         collections = list(collection_info.find({}))
-        for c in collections:
-            if c:
-                c.pop("_id", None)
-        return {"collections": collections}
-
+        if collections:
+            # added existence check here to match memory backend behavior
```

Comment:
Shouldn't need this. Iterating over an empty list is fine and has the desired behavior.

Reply:
The if-statement prevents the dict ( {"collections": collections} ) from being returned if there are no collections.  Before, the dict would always be returned throwing off the get_collections() view as the view checks if the dict is None in order to throw a 404. Otherwise the view would return 200 but with empty collections.

So if we dont like that logic, we need to make the backends consistent in their response via other   logic.

Reply:
Ah, I understand. I would just add an explicit `return None` at the end then (I missed the reliance on the default return value when the `if collections:` block isn't executed. 

Ideally, though (and this comes without a comprehensive understanding of how this works, so I could be missing important trade-offs and decisions, so take with a grain of salt), `Backend.get_collections` would just return a list of collections, not a dictionary with a `"collections"` key.

But more accurately, if there are no collections (but the API root does exist), the result should not be a 404, but a 200 with an empty JSON object (`{}`); the `"collections"` key [is optional](http://docs.oasis-open.org/cti/taxii/v2.0/cs01/taxii-v2.0-cs01.html#_Toc496542735).

Reply:
I noted your last point when looking over the function - Ill fix it. And make both backends behave as you describe in your second paragraph.